pkg/cache/clusterqueue_test.go

```diff
@@ -306,7 +381,32 @@ func TestFitInCohort(t *testing.T) {
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
 			defer features.SetFeatureGateDuringTest(t, features.LendingLimit, tc.enableLendingLimit)()
-			got := tc.cq.FitInCohort(tc.request)
+			cache := New(utiltesting.NewFakeClient())
+
+			for _, flavorName := range tc.flavorNames {
```

Comment:
nit: it looks like all the tests use "f1", and (most of the time) "f2"

Please delete `flavorNames` field, and just always create "f1" and "f2" flavors in the fixture

Reply:
I consider it problematic to pre-create `f1`, `f2` flavors in every test.
My understanding is that for `missing flavor` tests, only `f1` should be pre-created, not `f2`.
https://github.com/kubernetes-sigs/kueue/pull/2655/files#diff-e11e73ca517b5546e51925b6e6299ff79841efda37d39243ccee269d731de277L209-L228

Without creating `flavorNames`, it is not possible to create only `f1` in the `missing flavor` test. I think that `flavorNames` should remain in the implementation. What do you think?

Reply:
We can change `f2` in the `missing flavor` test to `non-existent-flavor`. This should also make the rest more readable, as it is immediately clear which flavor is missing.

The point of this suggestion is to be able to express the test data as concisely as possible, so readers can understand the test cases faster without having to filter out noise. In this case, the flavor doesn't add any information to the test suite.

Reply:
fix: https://github.com/kubernetes-sigs/kueue/pull/2655/commits/4457656656c8791572e8e256a9bd99f808096005